crates/flux-middle/src/rty/mod.rs

```diff
@@ -1536,6 +1536,37 @@ pub struct AliasTy {
 
 pub type RefineArgs = List<Expr>;
 
+#[extension(pub trait RefineArgsExt)]
+impl RefineArgs {
+    fn identity_for_item<F>(
+        genv: &GlobalEnv,
+        def_id: DefId,
+        mut mk_expr: F,
+    ) -> QueryResult<RefineArgs>
+    where
+        F: FnMut(usize, RefineParam) -> Expr,
+    {
+        let reft_generics = genv.refinement_generics_of(def_id)?;
+        let mut args = vec![];
+        for i in 0..reft_generics.count() {
+            let param = reft_generics.param_at(i, *genv)?;
+            args.push(mk_expr(i, param));
+        }
+        Ok(List::from_vec(args))
+    }
+    fn identity_for_item_new(genv: &GlobalEnv, def_id: DefId) -> QueryResult<RefineArgs> {
+        let reft_generics = genv.refinement_generics_of(def_id)?;
+        let mut args = vec![];
+        for i in 0..reft_generics.count() {
+            let param = reft_generics.param_at(i, *genv)?;
+            let expr =
+                Expr::var(Var::EarlyParam(EarlyReftParam { index: i as u32, name: param.name }));
+            args.push(expr);
+        }
+        Ok(List::from_vec(args))
+    }
+}
```

Comment:
I believe we only use `identity_for_item_new` 

```suggestion
    fn identity_for_item(genv: &GlobalEnv, def_id: DefId) -> QueryResult<RefineArgs> {
        let reft_generics = genv.refinement_generics_of(def_id)?;
        let mut args = vec![];
        for i in 0..reft_generics.count() {
            let param = reft_generics.param_at(i, *genv)?;
            let expr =
                Expr::var(Var::EarlyParam(EarlyReftParam { index: i as u32, name: param.name }));
            args.push(expr);
        }
        Ok(List::from_vec(args))
    }
}
```

If we need them both, we should implement one in terms of the other. Or we can have a `for_item` and an `identity_for_item`, both implemented on top of `fill_item` as we do for `GenericArgs`.